src/Microsoft.OData.Client/DeepInsertSaveResult.cs

```diff
@@ -431,24 +432,23 @@ private void HandleDeepInsertResponseInternal(MaterializerEntry entry, bool isTo
                         {
                             for (int i = 0; i < feed.Items.Count; i++)
                             {

```

Comment:
This loop handles all nested resources for a particular resource at that level.. Like if we have Product and product has Order and Customer related navigation properties.. It means we'll handle the Order and Customer at this level. What if Order and Customer had navigation properties? Could you add a test for that too?

Reply:
`foreach (IMaterializerState nestedItem in entry?.NestedItems)`
This is already part of the test as Manufactures and owners are both on the same level - the Car.

`for (int i = 0; i < feed.Items.Count; i++)`
This is when there a re multiple entries of the same multi navigation property. I can add another car to the test to showcase multiple cars on the same person.

Reply:
ok. I think I meant if Manufacturers and Owners had some nesting...like the nesting can go as deep as possible. There is no limit.